posthog/temporal/batch_exports/spmc.py

```diff
@@ -589,16 +597,84 @@ async def produce_batch_export_record_batches_from_range(
         done_ranges: collections.abc.Sequence[tuple[dt.datetime, dt.datetime]],
         queue: RecordBatchQueue,
         query_parameters: dict[str, typing.Any],
+        max_record_batch_size_bytes: int = 0,
+        min_records_per_batch: int = 100,
     ):
+        """Produce Arrow record batches for a given date range into `queue`.
+
+        Arguments:
+            query: The ClickHouse query used to obtain record batches. The query should be have a
+                `FORMAT ArrowStream` clause, although we do not enforce this.
+            full_range: The full date range of record batches to produce.
+            done_ranges: Date ranges of record batches that have already been exported, and thus
+                should be skipped.
+            queue: The queue where to produce record batches.
+            query_parameters: Additional query parameters.
+            max_record_batch_size_bytes: The max size in bytes of a record batch to insert in `queue`.
+                If a record batch is larger than this, `slice_record_batch` will be used to slice it
+                into smaller record batches.
+            min_records_batch_per_batch: If slicing a record batch, each slice should contain at least
```

Comment:
just checking why this argument is needed? just to ensure it is never too inefficient?

Reply:
Yeah, just adding a lower bound for safety. I don't think we would ever want to slice this down to a single record per batch. At that point it's no longer a batch.

Reply:
I do agree that 100 is a bit of an arbitrary default. Ideally we would look at some average weight per event in our database, and set a better default based on that.

Reply:
well I suppose unless there are really large records and the `max_record_batch_size_bytes` is really small then it shouldn't ever be an issue anyway